### PR TITLE
Rational interpolation

### PR DESCRIPTION
**Pull request by baruchel**
I adapted the RINTERP algorithm from the Axiom/FriCAS computer algebra system desbribed at:
http://axiom-wiki.newsynthesis.org/RationalInterpolation
and created the Sympy `rational_interpolate function` in `polyfuncs.py`

### CODE REVIEW

**Comment by baruchel** [comment]
There was a bug in my first version; I first thought that nullspace() was behaving strangely, which is why I tried a hack (bb64787). After having looked a second time, I noticed an issue with an index in my function. Now it is fixed (8158654). I am not sure whether I have to make again "a pull request" because it is my first pull request. However I see my following corrections; thus I assume I don't have to make a new pull request.

**Comment by gxyd** [comment]
> I am not sure whether I have to make again "a pull request"

You don't need to make a new PR. You can just update this one.

**Comment by baruchel** [comment]
Hi, as previously said, this is my first pull request; what does mean the "All checks have failed" above? Did I make something wrong?

**Comment by jksuom** [comment]
There seems to be a problem with doctests. You can find that by running ``bin/doctest polyfuncs``. It may be possible to fix that by appending a backslash to the line. There is also some extra whitespace that is revealed by ``bin/test quality``.

File: sympy/polys/polyfuncs.py [hunk 1]
```diff
@@ -255,6 +255,84 @@ def interpolate(data, x):
...
+    (see examples below). The  ``degnum`` argument is the degree in the
+    numerator of the rational function. Setting it too high will decrease the
+    maximal degree in the denominator for the same amount of data.
+
+    Example:
+    ========
+    >>> from sympy.polys.polyfuncs import rational_interpolate
+
+    A list is interpreted as though it were paired with a range starting
+    from 1:
```
Comment (asmeurer) [inline sympy/polys/polyfuncs.py]: Why starting with 1?

File: sympy/polys/polyfuncs.py [hunk 2]
```diff
@@ -255,6 +255,84 @@ def interpolate(data, x):
 
 
 @public
+def rational_interpolate(data, degnum, X='x'):
```
Comment (asmeurer) [inline sympy/polys/polyfuncs.py]: X should just be a Symbol, not a string.

**Comment by asmeurer** [comment]
I don't see the value of the multiple types of inputs. I would just support the list of tuples variant. If you have a dictionary you can use `.items` and if you have a list of y values you can use `zip` to easily convert to this type.

**Comment by baruchel** [comment]
I don't understand; why did the checks failed. As far as I can understand, the test that failed has nothing to do with the files and functions involved by my submission. What should I do?

**Comment by jksuom** [comment]
It is a PyPy issue. I'll restart that job.

**Comment by baruchel** [comment]
I just removed some spaces, and (again) a Pypy test failed on something that doesn't seem to be related with my commit. Could you restart this test?

**Comment by certik** [comment]
That looks good to me. Thanks @baruchel!

**Pull request merged**